bootstrap-salt.sh

```diff
@@ -1244,18 +1244,40 @@ __gather_system_info() {
 
 
 #---  FUNCTION  -------------------------------------------------------------------------------------------------------
-#          NAME:  __get_dpkg_architecture
-#   DESCRIPTION:  Determine primary architecture for packages to install on Debian and derivatives
+#          NAME:  __check_dpkg_architecture
+#   DESCRIPTION:  Determine the primary architecture for packages to install on Debian and derivatives
+#                 and issue all necessary error messages.
 #----------------------------------------------------------------------------------------------------------------------
-__get_dpkg_architecture() {
+__check_dpkg_architecture() {
     if __check_command_exists dpkg; then
         DPKG_ARCHITECTURE="$(dpkg --print-architecture)"
     else
         echoerror "dpkg: command not found."
         return 1
     fi
 
-    return 0
+    __REPO_ARCH="$DPKG_ARCHITECTURE"
+
+    if [ "$DPKG_ARCHITECTURE" = "i386" ]; then
+        echoerror "$_REPO_URL likely doesn't have all required 32-bit packages for Ubuntu $DISTRO_MAJOR_VERSION (yet?)."
```

Comment:
I guess we should use `${DISTRO_NAME}` variable instead of hard coded `Ubuntu` word here, right?